src/xbot/common/controls/sensors/JoystickButtonManager.java

```diff
@@ -44,14 +52,28 @@ public void addAnalogButton(AnalogHIDDescription desc) {
     }
 
     public AdvancedJoystickButton getifAvailable(int buttonNumber) {
+        
+        if (buttonNumber < 1 || buttonNumber > maxButtons) {
+            String message = "button " + buttonNumber + " is out of range!";
```

Comment:
I think we can streamline this to one line -- it seems like it would be easier to read.